tests/test_components/test_boundaries.py

```diff
@@ -188,3 +191,51 @@ def test_boundaryspec_classmethods():
     assert all(
         isinstance(boundary, PML) for boundary_dim in boundaries for boundary in boundary_dim
     )
+
+
+def test_pml_validator():
+    """Test the validators in class ``PML``"""
+    with pytest.raises(pydantic.ValidationError):
+        _ = PML(num_layers=0)
+
+
+@pytest.mark.parametrize(
+    "num_layers, log_level",
+    [(MIN_NUM_PML_LAYERS, None), (MIN_NUM_PML_LAYERS - 1, "WARNING")],
+)
+def test_pml_validator_warning(num_layers, log_level):
+    """Test the validators in ``PML`` which should display a warning not an error."""
+    with AssertLogLevel(log_level):
+        _ = PML(num_layers=num_layers)
+
+
+def test_stable_pml_validator():
+    """Test the validators in class ``StablePML``"""
+    with pytest.raises(pydantic.ValidationError):
+        _ = StablePML(num_layers=0)
+
+
+@pytest.mark.parametrize(
```

Comment:
you can nest parameterize decorators, eg if you want to loop through multiple dimensions. as an exercise why dont you I've it a try, looping over the 3 classes and their corresponding minimum layers values so these 3 tests become one